erpnext/manufacturing/doctype/bom/bom.py

```diff
@@ -140,6 +141,14 @@ def validate(self):
 		if not self.company:
 			frappe.throw(_("Please select a Company first."), title=_("Mandatory"))
 
+		items_in_msg = []
+		for idx, scrap_item in enumerate(self.scrap_items):
+			if isinstance(scrap_item.stock_qty, float) and frappe.db.get_value("UOM", scrap_item.stock_uom, "must_be_whole_number"):
+				self.scrap_items[idx].stock_qty = trunc(scrap_item.stock_qty)
+				items_in_msg.append(f"<b>{scrap_item.item_name}</b>")
+		if items_in_msg:
+			frappe.msgprint("Scrap cannot be in fraction for Item(s): " + ", ".join(map(str, items_in_msg)))
```

Comment:
```suggestion
			frappe.msgprint("Scrap rounded down since it cannot be a fraction for Item(s): " + ", ".join(map(str, items_in_msg)))
```
Also make this a toast by passing flags: 
```py 
alert=True, indicator='orange'
```